File: app/src/main/java/com/wolcano/musicplayer/music/ui/adapter/statepager/LibraryFragmentPagerAdapter.java
package com.wolcano.musicplayer.music.ui.adapter.statepager;

import android.content.Context;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.wolcano.musicplayer.music.R;
import com.wolcano.musicplayer.music.ui.fragment.library.FragmentAlbums;
import com.wolcano.musicplayer.music.ui.fragment.library.FragmentArtists;
import com.wolcano.musicplayer.music.ui.fragment.library.FragmentGenres;
import com.wolcano.musicplayer.music.ui.fragment.library.FragmentSongs;


public class LibraryFragmentPagerAdapter extends FragmentStatePagerAdapter {

    private Context context;

    public LibraryFragmentPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0) {
            return new FragmentSongs();
        } else if (position == 1) {
            return new FragmentAlbums();
        } else if (position == 2) {
            return new FragmentArtists();
        } else {
            return new FragmentGenres();
        }
    }


    @Override
    public int getCount() {
        return 4;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return context.getString(R.string.songs);
            case 1:
                return context.getString(R.string.albums);
            case 2:
                return context.getString(R.string.artists);
            case 3:
                return context.getString(R.string.genres);
            default:
                return null;
        }
    }

}

File: app/src/main/java/com/wolcano/musicplayer/music/ui/adapter/statepager/LibraryFragmentPagerAdapter.kt
package com.wolcano.musicplayer.music.ui.adapter.statepager

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.wolcano.musicplayer.music.R
import com.wolcano.musicplayer.music.ui.fragment.library.album.AlbumFragment
import com.wolcano.musicplayer.music.ui.fragment.library.artist.ArtistFragment
import com.wolcano.musicplayer.music.ui.fragment.library.genre.GenreFragment
import com.wolcano.musicplayer.music.ui.fragment.library.song.SongFragment

class LibraryFragmentPagerAdapter(private val context: Context, fm: FragmentManager) :
    FragmentStatePagerAdapter(
        fm
    ) {

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            SongFragment()
        } else if (position == 1) {
            AlbumFragment()
        } else if (position == 2) {
            ArtistFragment()
        } else {
            GenreFragment()
        }
    }

    override fun getCount(): Int {
        return 4
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> context.getString(R.string.songs)
            1 -> context.getString(R.string.albums)
            2 -> context.getString(R.string.artists)
            3 -> context.getString(R.string.genres)
            else -> null
        }
    }

}